File: Tutorials and Problems/P2c/Solutions c/LunchList3b/app/src/main/java/org/feup/apm/lunchlist3b/DetailsActivity.java
package org.feup.apm.lunchlist3b;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.RadioGroup;

public class DetailsActivity extends AppCompatActivity {
  LunchApp app;
  int rPos;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    ActionBar bar = getSupportActionBar();
    if (bar != null) {
      bar.setIcon(R.drawable.rest_icon);
      bar.setDisplayShowHomeEnabled(true);
    }
    setContentView(R.layout.activity_details);
    app = (LunchApp)getApplicationContext();
    findViewById(R.id.bt_save).setOnClickListener((View v)->onBtSaveClick());

    rPos=getIntent().getIntExtra(MainActivity.ID_EXTRA, -1);
    if (rPos != -1)
      load();
  }

  void onBtSaveClick() {
    String type = "";

    switch (((RadioGroup)findViewById(R.id.rg_types)).getCheckedRadioButtonId()) {
      case R.id.sit:
        type = "sit";
        break;
      case R.id.take:
        type="take";
        break;
      case R.id.delivery:
        type="delivery";
        break;
    }
    if (rPos!=-1)
      app.adapter.remove(app.current);
    app.current = new Restaurant();
    app.current.setName(((EditText)findViewById(R.id.ed_name)).getText().toString());
    app.current.setAddress(((EditText)findViewById(R.id.ed_address)).getText().toString());
    app.current.setNotes(((EditText)findViewById(R.id.ed_notes)).getText().toString());
    app.current.setType(type);
    if (rPos!=-1)
      app.adapter.insert(app.current, rPos);
    else
      app.adapter.add(app.current);
    finish();
  }

  void load() {
    ((EditText)findViewById(R.id.ed_name)).setText(app.current.getName());
    ((EditText)findViewById(R.id.ed_address)).setText(app.current.getAddress());
    ((EditText)findViewById(R.id.ed_notes)).setText(app.current.getNotes());
    RadioGroup rgTypes = findViewById(R.id.rg_types);
    if (app.current.getType().equals("sit"))
      rgTypes.check(R.id.sit);
    else if (app.current.getType().equals("take"))
      rgTypes.check(R.id.take);
    else
      rgTypes.check(R.id.delivery);
  }
}

File: Tutorials and Problems/3.3. The Lunch Restaurants App/LunchList3b/app/src/main/java/org/feup/apm/lunchlist3b/DetailsActivity.kt
package org.feup.apm.lunchlist3b

import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.RadioGroup
import androidx.appcompat.app.AppCompatActivity

class DetailsActivity : AppCompatActivity() {
  private val app by lazy { application as LunchApp }
  private val edName by lazy { findViewById<EditText>(org.feup.apm.lunchlist3b.R.id.ed_name) }
  private val edAddress by lazy { findViewById<EditText>(org.feup.apm.lunchlist3b.R.id.ed_address) }
  private val edNotes by lazy { findViewById<EditText>(org.feup.apm.lunchlist3b.R.id.ed_notes) }
  private val rgTypes by lazy { findViewById<RadioGroup>(org.feup.apm.lunchlist3b.R.id.rg_types) }
  private val rPos by lazy { intent.getIntExtra(ID_EXTRA, -1) }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    supportActionBar?.setIcon(R.drawable.rest_icon)
    supportActionBar?.setDisplayShowHomeEnabled(true)
    setContentView(R.layout.activity_details)

    findViewById<View>(R.id.bt_save).setOnClickListener { _ -> onBtSaveClick() }
    if (rPos != -1) load()
  }

  private fun onBtSaveClick() {
    val type = when(rgTypes.checkedRadioButtonId) {
      R.id.rb_take -> "take"
      R.id.rb_sit -> "sit"
      R.id.rb_delivery -> "delivery"
      else -> ""
    }
    if (rPos != -1)
      app.adapter!!.remove(app.current)
    app.current = Restaurant(edName.text.toString(), edAddress.text.toString(), type, edNotes.text.toString())
    if (rPos != -1)
      app.adapter!!.insert(app.current, rPos)
    else
      app.adapter!!.add(app.current)
    finish()
  }

  private fun load() {
    edName.setText(app.current?.name)
    edAddress.setText(app.current?.address)
    edNotes.setText(app.current?.notes)
    when (app.current?.type) {
      "sit" -> rgTypes.check(R.id.rb_sit)
      "take" -> rgTypes.check(R.id.rb_take)
      "delivery" -> rgTypes.check(R.id.rb_delivery)
    }
  }
}